core/variant_op.cpp

```diff
@@ -406,8 +406,6 @@ void Variant::evaluate(const Operator &p_op, const Variant &p_a,
 
 			CASE_TYPE(math, OP_EQUAL, BOOL) {
 				if (p_b.type != BOOL) {
-					if (p_b.type == NIL)
-						_RETURN(false);
```

Comment:
Are you sure about this change? It reads like previously `true == null` would return `false` (tested, confirmed), while now it would fail.

Reply:
For me this also was strange but static analyser shows that that this can be a bug.
I don't know how to check this but this code:
```
print(true == null)
print(null == true)
print(null == false)
print(false == null)
```
return 4 times False on master and this branch, so I think that analyser can have a right

Reply:
What was the report from the static analyzer? I don't see it in any of the four referenced issues.

Reply:
This one really does not make sense to me, i would not touch it.

Reply:
I found it recently and waited to collect more to create another issue.
This is an message from analyser
```
godot/core/variant_op.cpp:409: ostrzeżenie: V547 Expression 'p_b.type == NIL' is always false.
```
also similar part of code have another potentially issue
```
		SWITCH_OP(math, OP_NOT_EQUAL, p_a.type) {
			CASE_TYPE(math, OP_NOT_EQUAL, NIL) {
				if (p_b.type == NIL) _RETURN(false);
				if (p_b.type == OBJECT)
					_RETURN(p_b._get_obj().obj != NULL);

				_RETURN(true);
			}
```
and this is message
```
godot/core/variant_op.cpp:486: ostrzeżenie: V779 Unreachable code detected. It is possible that an error is present.
```